packages/destination-actions/src/destinations/sendgrid/updateUserProfile/index.ts

```diff
@@ -0,0 +1,251 @@
+      }
+    },
+    address_line_1: {
+      label: 'Address Line 1',
+      description: `The user's address line 1`,
+      type: 'string',
+      allowNull: true,
+      default: {
+        '@if': {
+          exists: { '@path': '$.traits.address.line1' },
+          then: { '@path': '$.traits.address.line1' },
+          else: { '@path': '$.properties.address.line1' }
+        }
+      }
+    },
+    address_line_2: {
+      label: 'Address Line 2',
+      description: `The user's address line 2`,
+      type: 'string',
+      allowNull: true,
+      default: {
+        '@if': {
+          exists: { '@path': '$.traits.address.line2' },
+          then: { '@path': '$.traits.address.line2' },
+          else: { '@path': '$.properties.address.line2' }
+        }
+      }
+    },
+    phone_number: {
+      label: 'Phone Number',
+      description: `The user's phone number`,
+      type: 'string',
+      allowNull: true,
+      default: {
+        '@if': {
+          exists: { '@path': '$.traits.phone_number' },
+          then: { '@path': '$.traits.phone_number' },
+          else: { '@path': '$.properties.phone_number' }
+        }
+      }
+    },
+    whatsapp: {
+      label: 'whatsapp',
+      description: `The user's whatsapp`,
+      type: 'string',
+      allowNull: true,
+      default: {
+        '@if': {
+          exists: { '@path': '$.traits.whatsapp' },
+          then: { '@path': '$.traits.whatsapp' },
+          else: { '@path': '$.properties.whatsapp' }
+        }
+      }
+    },
+    line: {
+      label: 'Line Id',
+      description: `The user's line id`,
+      type: 'string',
+      allowNull: true,
+      default: {
+        '@if': {
+          exists: { '@path': '$.traits.line' },
+          then: { '@path': '$.traits.line' },
+          else: { '@path': '$.properties.line' }
+        }
+      }
+    },
+    facebook: {
+      label: 'Facebook Id',
+      description: `The user's facebook id`,
+      type: 'string',
+      allowNull: true,
+      default: {
+        '@if': {
+          exists: { '@path': '$.traits.facebook' },
+          then: { '@path': '$.traits.facebook' },
+          else: { '@path': '$.properties.facebook' }
+        }
+      }
+    },
+    unique_name: {
+      label: 'UniqueName',
+      description: `The user's UniqueName`,
+      type: 'string',
+      allowNull: true,
+      default: {
+        '@if': {
+          exists: { '@path': '$.traits.unique_name' },
+          then: { '@path': '$.traits.unique_name' },
+          else: { '@path': '$.properties.unique_name' }
+        }
+      }
+    },
+    identity: {
+      label: 'Identity',
+      description: `The user's Identity`,
+      type: 'string',
+      allowNull: true,
+      default: {
+        '@if': {
+          exists: { '@path': '$.traits.identity' },
+          then: { '@path': '$.traits.identity' },
+          else: { '@path': '$.properties.identity' }
+        }
+      }
+    },
+    primary_email: {
+      label: 'Email',
```

Comment:
Since this is required as per the API docs, we should probably add `required: true` and that will also result in this field being automatically show at the top of the list in the UI.

```suggestion
      label: 'Email Address',
```